packages/core/src/validate.js

```diff
@@ -263,15 +264,60 @@ export default function validateFormData(
   };
 }
 
+/**
+ * Get a similar schema where ref's are prefixed with `rootSchemaId`
+ * This is used in isValid to make references to the rootSchema
+ */
+function withIdRefPrefix(schemaNode) {
+  let obj = schemaNode;
+  if (schemaNode.constructor === Object) {
+    obj = { ...schemaNode };
+    for (let key of Object.keys(obj)) {
+      const value = obj[key];
+      if (key === "$ref") {
+        obj[key] = rootSchemaId + value;
+      } else {
+        obj[key] = withIdRefPrefix(value);
+      }
+    }
+  } else if (Array.isArray(schemaNode)) {
+    obj = [...schemaNode];
+    for (var i = 0; i < obj.length; i++) {
+      obj[i] = withIdRefPrefix(obj[i]);
+    }
+  }
+  return obj;
+}
+
 /**
  * Validates data against a schema, returning true if the data is valid, or
  * false otherwise. If the schema is invalid, then this function will return
  * false.
  */
-export function isValid(schema, data) {
+export function isValid(schema, data, rootSchema) {
   try {
-    return ajv.validate(schema, data);
+    // if rootSchema is given then we add that schema with an id.
+    // then we rewrite the schema ref's to point to the rootSchema using the id
+    // this accounts for the case where schema have references to models
+    // that lives in the rootSchema but not in the schema in question.
+    if (rootSchema) {
+      const result = ajv
+        .addSchema(rootSchema, rootSchemaId)
+        .validate(withIdRefPrefix(schema), data);
+
+      // make sure we remove the rootSchema from the global ajv instance
+      ajv.removeSchema(rootSchemaId);
+      return result;
+    } else {
+      return ajv.validate(schema, data);
+    }
   } catch (e) {
+    try {
+      // make sure we also remove the rootSchema if an error occured before removing but after adding
+      ajv.removeSchema(rootSchemaId);
+    } catch (e) {
+      return false;
+    }
     return false;
   }
```

Comment:
To make this cleaner, and consolidate the removeSchema logic above, let's actually just remove the schema in a finally block:

```suggestion
  }
  finally {
    try {
      ajv.removeSchema(rootSchemaId);
    } ... 
  }
```

Also, do you know if `removeSchema` returns an error if `rootSchemaId` hasn't already been added to ajv?

Reply:
Oh it doesn't raise an error so no need for try around that part